lib/bandit/http1/adapter.ex

```diff
@@ -106,6 +112,21 @@ defmodule Bandit.HTTP1.Adapter do
   defp get_version({1, 0}), do: {:ok, :"HTTP/1.0"}
   defp get_version(other), do: {:error, "invalid HTTP version: #{inspect(other)}"}
 
+  defp get_content_length(headers) do
+    with {_, value} <- List.keyfind(headers, "content-length", 0),
+         [length] <- Enum.uniq(Plug.Conn.Utils.list(value)),
```

Comment:
(Apologies for the code golf here but this is a *really* hot path and I want to make sure it's as fast as possible).

I wonder if we should be special-casing the (almost universally common) case where we have a single content-length value, and avoiding the uniq call. I looked into the Enum.uniq implementation and it's (necessarily) complex, even for lists of length 1. Maybe something like 
```suggestion
         {:ok, length} <- enforce_unique_value(value),
```
along with:
```elixir
defp enforce_unique_value([value]), do: {:ok, value}
defp enforce_unique_value(value) do
  case Enum.uniq(Plug.Conn.Utils.list(value)) do
    [value] -> {:ok, value}
    _ -> {:error, "???"}
  end
end
```

I ran a quick benchee to see the difference, and it's quite staggering:

```
defmodule Foo do
  def uniq do
    [1] 
    |> Enum.uniq 
    |> case do
      [x] -> x
    end
  end

  def single do
    [1] 
    |> case do
      [x] -> x
    end
  end
end

Mix.install([{:benchee, "~> 1.0", only: :dev}])

Benchee.run(%{
  uniq: &Foo.uniq/0,
  single: &Foo.single/0
})


Name             ips        average  deviation         median         99th %
single      338.16 M        2.96 ns   ±334.86%        2.96 ns        3.08 ns
uniq         11.22 M       89.11 ns ±25896.88%          83 ns          84 ns

Comparison:
single      338.16 M
uniq         11.22 M - 30.13x slower +86.15 ns
```




WDYT?

Reply:
I updated the PR and removed the call to `Enum.uniq/1` completely in favor of just checking the values recursively (since it can only be a list). If it is only one value, the function returns early.
We still need the call to [Plug.Conn.Utils.list/1](https://github.com/elixir-plug/plug/blob/d7940552e365e0d3245bb78ca99429ea5c324f7d/lib/plug/conn/utils.ex#L270) though, which could impact performance.